src/AutoComplete/AutoComplete.component.tsx

```diff
@@ -16,38 +16,39 @@ type AutoCompleteDataSource = Array<{
 }>;
 
 interface AutoCompleteProps {
+    className?: string;
+    dataSource?: AutoCompleteDataSource | string[] | number[];
     debug?: boolean;
     name?: string;
-    dataSource?: AutoCompleteDataSource | string[] | number[];
-    className?: string;
-    size?: 'sm' | 'md' | 'lg';
     placeholder?: string;
+    size?: 'sm' | 'md' | 'lg';
     value?: string | number;
     // TODO: resultTemplate
     // TODO: allowClear
     allowClear?: boolean;
     // Visual
-    border?: boolean;
-    shadow?: boolean;
     background?: string;
+    border?: boolean;
     color?: string;
+    shadow?: boolean;
     // TODO: Allow children
     resultTemplate?: (props: any) => any;
-    suffix?: any;
+    inputComponent?: any;
```

Comment:
Still not clear on this prop. Have you seen [this](https://www.styled-components.com/docs/api#as-polymorphic-prop)?

Reply:
The `inputComponent` prop allows the user to provide their own JSX to replace the `Input` component being used. Theoretically any JSX could be supplied, but realistically it allows a new `Input` to be supplied with whatever attributes the user wants to tack on. Line `168` has an example of the usage.

I looked into `as` and I'm not exactly sure how that would solve this. I tried `<AutoComplete as="input" autoCapitalize="false />` and get TS errors so I'm guessing that's not what you're suggesting.

Reply:
I tried using `as` and passing the custom component into it. The problem is that the `ref` gets lost when doing so. I've tried a variety of ways to pass it along including the `forwardedRef` prop from `styled-components` with no luck.